2-Authorization-I/1-call-graph/SPA/src/app/account-switch/account-switch.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import {
+  MsalService,
+  MSAL_GUARD_CONFIG,
+  MsalGuardConfiguration,
+} from '@azure/msal-angular';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import {
+  RedirectRequest,
+  InteractionRequiredAuthError,
+  SilentRequest,
+  AccountInfo,
+} from '@azure/msal-browser';
+
+@Component({
+  selector: 'app-account-switch-component',
+  templateUrl: './account-switch.component.html',
+  styleUrls: ['./account-switch.component.css'],
+})
+export class AccountSwitchComponent implements OnInit {
+  accounts: AccountInfo[] = [];
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    @Inject(MSAL_GUARD_CONFIG) private msalGuardConfig: MsalGuardConfiguration,
+    private authService: MsalService
+  ) {}
+
+  ngOnInit(): void {}
+
+  switchAccount(account: AccountInfo | null) {
+    const activeAccount = this.authService.instance.getActiveAccount();
+    if (!account) {
+      this.authService.instance.setActiveAccount(account);
+      this.authService.instance.loginRedirect({
+        ...this.msalGuardConfig.authRequest,
+        prompt: 'login',
+      } as RedirectRequest);
+    } else if (
+      account &&
+      activeAccount?.homeAccountId != account.homeAccountId
```

Comment:
```suggestion
      activeAccount?.homeAccountId != account.homeAccountId
```
can we use strict equality here?